pages/api/create-checkout-session.ts

```diff
@@ -1,18 +1,20 @@
-import { stripe } from '@/utils/stripe';
-import { getUser } from '@/utils/supabase-admin';
-import { createOrRetrieveCustomer } from '@/utils/useDatabase';
-import { getURL } from '@/utils/helpers';
+import { stripe } from 'utils/stripe';
+import { getUser } from 'utils/supabase-admin';
+import { createOrRetrieveCustomer } from 'utils/useDatabase';
+import { getURL } from 'utils/helpers';
+import { NextApiRequest, NextApiResponse } from 'next';
 
-const createCheckoutSession = async (req, res) => {
+const createCheckoutSession = async (req: NextApiRequest, res: NextApiResponse) => {
   if (req.method === 'POST') {
-    const token = req.headers.token;
+    const token = req.headers.token as string;
```

Comment:
Headers are of type `IncomingHttpHeaders`, which extends `NodeJS.Dict<string | string[]>`, meaning this could ultimately be `string | string[] | undefined`, we're counting on the token being correctly sent as a string.